### PR TITLE
Support Defining Globals for Instructions

### PR DESCRIPTION
**Pull request by SamirDroubi**

### CODE REVIEW

**Comment by SamirDroubi** [comment]
It is expected for the avx2 ExoBLAS codegen tests to fail since this commit also minimized the number of empty lines generated in the source files.

File: src/exo/LoopIR.py [hunk 1]
```diff
@@ -163,9 +166,12 @@ def __new__(cls, op):
                 fnarg*          args,
                 expr*           preds,
                 stmt*           body,
-                string?         instr,
+                instr?          instr,
                 srcinfo         srcinfo )
 
+    instr   = ( string          instr,
```
Comment (skeqiqevian) [inline src/exo/LoopIR.py:172]: Writing `instr.instr` is a little confusing, how about `instr.c_instr` or something like that?

Comment (SamirDroubi) [reply src/exo/LoopIR.py:172]: Yeah that's better. I will also rename `global_` to `c_global`.

Comment (SamirDroubi) [reply src/exo/LoopIR.py]: Done

**Review by yamaguchi1024 — CHANGES_REQUESTED** [review]
- you probably want a test which directly constructs `@instr(..)` with global includes
- also a test which includes some C stdlib headers

**Comment by yamaguchi1024** [comment]
@SamirDroubi What's the status of this PR? Is this waiting for me or Kevin's review?

**Comment by SamirDroubi** [comment]
@yamaguchi1024 yes

File: src/exo/API.py [hunk 2]
```diff
@@ -341,7 +341,7 @@ def c_code_str(self):
         return decls + "\n" + defns
 
     def compile_c(self, directory: Path, filename: str):
-        compile_procs([self._loopir_proc], directory, f"{filename}.c", f"{filename}.h")
+        compile_procs([self], directory, f"{filename}.c", f"{filename}.h")
```
Comment (yamaguchi1024) [inline src/exo/API.py:344]: I remember seeing this and we did talk about it but I forgot; was this a bug?

**Review by yamaguchi1024 — APPROVED** [review]

File: src/exo/API.py (hunk 2 above)
Comment (SamirDroubi) [reply src/exo/API.py:344]: It was untested code; someone changed the interface of compile_procs at some point and as a result this call was incorrect.

**Pull request merged**